sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,729 @@
+----
+
+_Effects_: Enqueues a commands to `q` or the SYCL queue associated with `h`
+which will asynchornously allocate memory of size `size` in bytes, allocating
+from the memory pool `pool` if provided, otherwise allocation from the default
+memory pool associated with the SYCL context and device associated with `q` or
+`h`. Memory is first allocated from the memory pool if possible, otherwise
+memory is allocated from the OS to the memory pool to provide enough memory in
+the memory pool for the allocation.
+
+_Returns_: A pointer to the address of a memory reservation.
+
+_Throws_: An exception with the `errc::invalid` error code if `size` is zero.
+An exception with the `errc::memory_allocation` error code if the allocation
+brings the memory pool over the maximum size. This error can be thrown
+asynchornously.
+
+[_Note:_ Accessing the memory at the address of the pointer returned by
+asynchornous malloc functions before the command has completed execution is
+undefined behaviour. _{endnote}_]
+
+[source, c++]
+----
+void async_free(queue q, void *ptr);
+
+void async_free(handler h, void *ptr);
+----
+
+_Effects_: Enqueues a commands to `q` or the SYCL queue associated with `h`
+which will asynchornously free the memory allocation at the address of `ptr`.
+Memory will be freed from the memory pool to be used by other asynchronous
+malloc commands which execute later, and will not free to the OS until the SYCL
+queue associated with the asycnhronous allocation command has been synchornized
+with.
+
+_Throws_: An exception with the `errc::invalid` error code if `ptr` is not the
+address of a memory allocation allocated to a memory pool.
+
+[_Note:_ Accessing the memory at the address of `ptr`after the asynchronous free
+command has completed execution is undefined behaviour. _{endnote}_]
+
+
+=== Memory pool lifetimes
+
+The lifetime of memory allocated via a memory pool is as follows.
+
+When an asynchornous malloc command is executed it will first look to the
+memory pool to opportunistically allocate memory already available there. The
+SYCL RT may also, when possible, look at all asynchronous malloc and free
+commands and their dependencies in order to more efficiently schedule memory
+allocations. If the memory pool does not have the requirement memory available
+to allocate for the asynchronous malloc command it will then look to allocate
+from the OS to the memory pool to provide the required memory.
+
+When an asynchornous free command is executed it will free up memory in the 
+memory pool to be used for a later asynchonrous malloc command. Memory is not
+freed back to the OS at this point.
+
+When a SYCL queue which enqueued asynchronous malloc commands is synchronized
+with, i.e. when `queue::wait` oo `queue::wait_and_throw` is called, the SYCL RT\
+will check the current total memory allocated after to the memory pool against
+the memory pool's deallocation threshold. If the total memory allocated islarger
```

Comment:
```suggestion
the memory pool's deallocation threshold. If the total memory allocated is larger
```

Reply:
Fixed in https://github.com/intel/llvm/pull/14800/commits/2c51adf28ece21c0e91c2c0ec85440ae1153e536.